MapboxDirections/MBDirectionOptions.swift

```diff
@@ -0,0 +1,449 @@
+import Foundation
+
+/**
+ A `RouteShapeFormat` indicates the format of a route’s shape in the raw HTTP response.
+ */
+@objc(MBRouteShapeFormat)
+public enum RouteShapeFormat: UInt, CustomStringConvertible {
+    /**
+     The route’s shape is delivered in [GeoJSON](http://geojson.org/) format.
+     
+     This standard format is human-readable and can be parsed straightforwardly, but it is far more verbose than `polyline`.
+     */
+    case geoJSON
+    /**
+     The route’s shape is delivered in [encoded polyline algorithm](https://developers.google.com/maps/documentation/utilities/polylinealgorithm) format with 1×10<sup>−5</sup> precision.
+     
+     This machine-readable format is considerably more compact than `geoJSON` but less precise than `polyline6`.
+     */
+    case polyline
+    /**
+     The route’s shape is delivered in [encoded polyline algorithm](https://developers.google.com/maps/documentation/utilities/polylinealgorithm) format with 1×10<sup>−6</sup> precision.
+     
+     This format is an order of magnitude more precise than `polyline`.
+     */
+    case polyline6
```

Comment:
One day, we'll have the ability to use associated-value enums.

`case polyline(version: FrameworkVersion)`

Reply:
Someday perhaps. Note that the “6” here refers to a precision (number of decimal places), not a version. Six is actually the usual value for the Polyline encoding, but OSRM v5 went with five places before adding a sixth as an option more recently for better interoperability.

Reply:
🤔 Hmm, TIL!